### ISSUE TITLE
XmlNamespaced attributes are malformed when run through XhtmlMarkupFormatter

### ISSUE THREAD

**Issue by jbrayfaithlife** [AngleSharp/AngleSharp#1115]
# Bug Report

Given an XML an attribute has the namespace `xmlns`, the resulting output from the XhtmlMarkupFormatter is malformed.

## Prerequisites

- [/] Can you reproduce the problem in a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)?
- [/] Are you running the latest version of AngleSharp?
- [/] Did you check the FAQs to see if that helps you?
- [/] Are you reporting to the correct repository? (there are multiple AngleSharp libraries, e.g., `AngleSharp.Css` for CSS support)
- [/] Did you perform a search in the issues?

For more information, see the `CONTRIBUTING` guide.

## Description

An XmlElement represented by the text `<span xmlns:type='jump'>Test</span>` returns the following `<span xmlns:="jump">Test</span>` when run through the XhtmlMarkupFormatter which is not semantically equivalent.

## Steps to Reproduce

```
var xmlParser = new XmlParser();
var doc = xmlParser.ParseDocument("<span xmlns:type='jump'>Test</span>");
using (var stringWriter = new StringWriter()){
	doc.ToHtml(stringWriter, new XhtmlMarkupFormatter());
	stringWriter.ToString().Dump();
}
```

**Expected behavior:** Dump should output `<span xmlns:type='jump'>Test</span>`

**Actual behavior:** Dump should output `<span xmlns:="jump">Test</span>`

**Environment details:** Win 10, .NET 6.0.15

## Possible Solution

I fixed this by subclassing the XhtmlMarkupFormatter and hiding the behavior of `XmlNamespaceLocalName` https://github.com/AngleSharp/AngleSharp/blob/ebf660279f9f4c74cbade95e38e7d7d93b74dac2/src/AngleSharp/Xhtml/XhtmlMarkupFormatter.cs#L207

and changing it to
```
public new static string XmlNamespaceLocalName(string name) => !name.Is(NamespaceNames.XmlNsPrefix) ? $"{NamespaceNames.XmlNsPrefix}:{name}" : name;
```

If this is correct, I can create a PR to this effect.

**Comment by FlorianRappl** [comment]
Seems to correct to me - as with the other issue #1114 I'd appreciate a PR, but we need to see if there are side-effects and (in case there are) how to avoid them.

**Comment by jbrayfaithlife** [comment]
Upon doing further research, I don't think this is a bug with the formatter, though changing the formatter got around the problem.

I *think* that the bug is actually in the `XmlDomBuilder` in the AngleSharp.Xml project. After further testing, I realized that the actual problem I found in the wild has to do with `xml` prefixed attributes. 

What is happening essentially is that in the context of an xml document the attributes using the `xml` prefix, such as `xml:lang` should be available without an explicit namespace declaration. However the parser goes to look for a registered namespace on an element, and does not find one. 

I reported https://github.com/AngleSharp/AngleSharp.Xml/issues/20 to cover my problem and this issue may be closed.
